### PR TITLE
Chasms spit out things that aren't deletable

### LINKED ISSUE #20979: The eternally falling talisman
http://i.imgur.com/jXAY4pW.png

So from what I could get from the player, the player dropped the talisman down the chasm and it just kept falling... And falling.... And falling. I couldn't get the vars for it because it doesn't exist.

### PR DESCRIPTION
**Pull request by silicons**
Fixes #20979 in a crude way.

### CODE REVIEW

File: code/game/turfs/simulated/chasm.dm [hunk 1]
```diff
@@ -139,6 +139,9 @@
 		qdel(S.mmi)
 
 	qdel(AM)
-
+	
+	if(AM && !QDELETED(AM))	//It's indestructible
+		AM.throw_at(get_edge_target_turf(src,pick(alldirs)),rand(1, 10),rand(1, 10))
```
Comment (Fox-McCloud) [inline code/game/turfs/simulated/chasm.dm]: `throw_at_fast` or else this will hold things up until

Comment (Cyberboss) [inline code/game/turfs/simulated/chasm.dm]: All throwing is async now

**Comment by Shadowlight213** [comment]
Should add a burp sound effect when it throws something out

File: code/game/turfs/simulated/chasm.dm (hunk 1 above)
Comment (ChangelingRain) [inline code/game/turfs/simulated/chasm.dm]: also you gotta update_atom_color and reset its alpha and pixel_y or your object will be invisible.

Comment (optimumtact) [inline code/game/turfs/simulated/chasm.dm]: @Cyberboss pls

Comment (optimumtact) [inline code/game/turfs/simulated/chasm.dm]: err @kevinz000

Comment (kevinz000) [inline code/game/turfs/simulated/chasm.dm]: oh shit honk.

File: code/game/turfs/simulated/chasm.dm [hunk 2]
```diff
@@ -139,6 +141,13 @@
 		qdel(S.mmi)
 
 	qdel(AM)
-
+	
+	if(AM && !QDELETED(AM))	//It's indestructible
+		visible_message("<span class='boldwarning'>[src] spits out the [AM]!</span>")
+		AM.alpha = 255
```
Comment (ChangelingRain) [inline code/game/turfs/simulated/chasm.dm]: where's your oldalpha?

File: code/game/turfs/simulated/chasm.dm [hunk 3]
```diff
@@ -144,7 +145,7 @@
 	
 	if(AM && !QDELETED(AM))	//It's indestructible
 		visible_message("<span class='boldwarning'>[src] spits out the [AM]!</span>")
-		AM.alpha = 255
+		AM.alpha = oldalpha
```
Comment (RemieRichards) [inline code/game/turfs/simulated/chasm.dm]: APPEARANCE CHURN 1

File: code/game/turfs/simulated/chasm.dm [hunk 4]
```diff
@@ -144,7 +145,7 @@
...
 		AM.color = oldcolor
```
Comment (RemieRichards) [inline code/game/turfs/simulated/chasm.dm]: APPEARANCE CHURN 2

File: code/game/turfs/simulated/chasm.dm [hunk 5]
```diff
@@ -144,7 +145,7 @@
...
 		AM.transform = oldtransform
```
Comment (RemieRichards) [inline code/game/turfs/simulated/chasm.dm]: APPEARANCE CHURN 3

You're all fired.

Comment (optimumtact) [inline code/game/turfs/simulated/chasm.dm]: nah it's fine this won't happen that often

Comment (RemieRichards) [inline code/game/turfs/simulated/chasm.dm]: NO EXCUSE!

Comment (RemieRichards) [inline code/game/turfs/simulated/chasm.dm]: the non-churny code is also 2 lines instead of 6 so.

**Pull request merged**